packages/api/src/folder/queries/get_folder_children.sql

```diff
@@ -0,0 +1,110 @@
+WITH all_children AS (
+  SELECT
+    id,
+    item_type,
+    folder_linkid,
+    row_number() OVER (
+      ORDER BY
+        (CASE
+          WHEN (
+            SELECT sort
+            FROM folder
+            WHERE folderid = :parentFolderId
+          ) = 'sort.alphabetical_asc'
+            THEN displayname
+        END) ASC,
+        (CASE
+          WHEN (
+            SELECT sort
+            FROM folder
+            WHERE folderid = :parentFolderId
+          ) = 'sort.alphabetical_desc'
+            THEN displayname
+        END) DESC,
+        (CASE
+          WHEN (
+            SELECT sort
+            FROM folder
+            WHERE folderid = :parentFolderId
+          ) = 'sort.display_date_asc'
+            THEN displaydt
+        END) ASC,
+        (CASE
+          WHEN (
+            SELECT sort
+            FROM folder
+            WHERE folderid = :parentFolderId
+          ) = 'sort.display_date_desc'
+            THEN displaydt
+        END) DESC,
+        (CASE
+          WHEN (
+            SELECT sort
+            FROM folder
+            WHERE folderid = :parentFolderId
+          ) = 'sort.type_asc'
+            THEN type
+        END) ASC,
+        (CASE
+          WHEN (
+            SELECT sort
+            FROM folder
+            WHERE folderid = :parentFolderId
+          ) = 'sort.type_desc'
+            THEN type
+        END) DESC
+    ) AS "rank"
+  FROM (
+    SELECT
+      folder_link.folderid AS "id",
+      'folder' AS "item_type",
+      folder_link.folder_linkid,
+      folder.displayname,
+      folder.displaydt,
+      folder.type
+    FROM
+      folder_link
+    INNER JOIN
+      folder
+      ON folder_link.folderid = folder.folderid
+    WHERE
+      folder_link.parentfolderid = :parentFolderId
+    UNION
+    SELECT
+      folder_link.recordid AS "id",
+      'record' AS "item_type",
+      folder_link.folder_linkid,
+      record.displayname,
+      record.displaydt,
+      record.type
+    FROM
+      folder_link
+    INNER JOIN
+      record
+      ON folder_link.recordid = record.recordid
+    WHERE
+      folder_link.parentfolderid = :parentFolderId
+  ) AS "children"
+),
+
+cursor AS (
+  SELECT rank
+  FROM
+    all_children
+  WHERE
+    folder_linkid = :cursor
+),
+
+total_pages AS (
+  SELECT ceiling(count(*) / :pageSize) AS "total_pages"
+  FROM all_children
+)
+
+SELECT
+  id,
+  item_type,
+  (SELECT total_pages FROM total_pages) AS "totalPages"
+FROM all_children
+WHERE
+  rank > coalesce((SELECT rank FROM cursor), 0)
+LIMIT :pageSize;
```

Comment:
Do we need an `order by rank` as part of this final select to ensure the sort order is honored in the actual result set?

(and if so, should we include a test that ensures order is honored in results?)

Reply:
The test exists, it's just currently a non-deterministic test that happens to have passed so far. Will add the `ORDER BY`.